fdb-record-layer-lucene/src/main/java/org/apache/lucene/codecs/lucene84/LuceneOptimizedPostingsReader.java

```diff
@@ -0,0 +1,2006 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.lucene.codecs.lucene84;
+
+import static org.apache.lucene.codecs.lucene84.ForUtil.BLOCK_SIZE;
+import static org.apache.lucene.codecs.lucene84.Lucene84PostingsFormat.DOC_CODEC;
+import static org.apache.lucene.codecs.lucene84.Lucene84PostingsFormat.MAX_SKIP_LEVELS;
+import static org.apache.lucene.codecs.lucene84.Lucene84PostingsFormat.PAY_CODEC;
+import static org.apache.lucene.codecs.lucene84.Lucene84PostingsFormat.POS_CODEC;
+import static org.apache.lucene.codecs.lucene84.Lucene84PostingsFormat.TERMS_CODEC;
+import static org.apache.lucene.codecs.lucene84.Lucene84PostingsFormat.VERSION_COMPRESSED_TERMS_DICT_IDS;
+import static org.apache.lucene.codecs.lucene84.Lucene84PostingsFormat.VERSION_CURRENT;
+import static org.apache.lucene.codecs.lucene84.Lucene84PostingsFormat.VERSION_START;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import com.apple.foundationdb.annotation.SpotBugsSuppressWarnings;
+import org.apache.lucene.codecs.BlockTermState;
+import org.apache.lucene.codecs.CodecUtil;
+import org.apache.lucene.codecs.PostingsReaderBase;
+import org.apache.lucene.codecs.lucene84.Lucene84PostingsFormat.IntBlockTermState;
+import org.apache.lucene.index.FieldInfo;
+import org.apache.lucene.index.Impacts;
+import org.apache.lucene.index.ImpactsEnum;
+import org.apache.lucene.index.IndexFileNames;
+import org.apache.lucene.index.IndexOptions;
+import org.apache.lucene.index.PostingsEnum;
+import org.apache.lucene.index.SegmentReadState;
+import org.apache.lucene.index.SlowImpactsEnum;
+import org.apache.lucene.store.DataInput;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.util.ArrayUtil;
+import org.apache.lucene.util.BitUtil;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.IOUtils;
+import org.apache.lucene.util.RamUsageEstimator;
+
+/**
+ * Concrete class that reads docId(maybe frq,pos,offset,payloads) list
+ * with postings format.
+ *
+ *
+ */
+@SuppressWarnings({"PMD.CompareObjectsWithEquals", "PMD.TooManyStaticImports", "PMD.FormalParameterNamingConventions", "PMD.UnnecessaryFullyQualifiedName"})
+@SpotBugsSuppressWarnings("BC_UNCONFIRMED_CAST")
+public final class LuceneOptimizedPostingsReader extends PostingsReaderBase {
+
+    private static final long BASE_RAM_BYTES_USED = RamUsageEstimator.shallowSizeOfInstance(Lucene84PostingsReader.class);
+
+    private final IndexInput docIn;
+    private final IndexInput posIn;
+    private final IndexInput payIn;
+
+    private final int version;
+
+    /** Sole constructor. */
+    public LuceneOptimizedPostingsReader(SegmentReadState state) throws IOException {
```

Comment:
Could you add comments to indicate that the difference between this optimized one and the ```Lucene84PostingsReader``` is the removed ```CodecUtil.retrieveChecksum```